R/readLPJmLClimateInput.R

```diff
@@ -0,0 +1,197 @@
+      x        <- collapseNames(as.magpie(x, spatial = 1))
+
+
+    } else if (subset == "annualSum") {
+
+      x <- lpjclass::read.LPJ_input(file_name   = filename,
+                                    out_years   = paste0("y", years),
+                                    namesum     = TRUE,
+                                    ncells      = 67420)
+
+      class(x) <- "array"
+      x        <- collapseNames(as.magpie(x, spatial = 1))
+
+
+    } else if (subset %in% c("monthlyMean", "monthlySum")) {
+      # define year sets (cut it in bunches)
+      bunchLength <- 1
+      yearsets    <- split(years, ceiling(seq_along(years) / bunchLength))
+
+      # define month mapping
+      monthLength <- c(jan = 31, feb = 28, mar = 31, apr = 30,
+                       may = 31, jun = 30, jul = 31, aug = 31,
+                       sep = 30, oct = 31, nov = 30, dec = 31)
+      daysMonth   <- NULL
+      for (m in 1:12) {
+        daysMonth <- c(daysMonth, rep(names(monthLength[m]),
+                                      monthLength[m]))
+      }
+      month2day   <- cbind(day   = 1:sum(monthLength),
+                           month = daysMonth)
+      monthLength <- as.magpie(monthLength)
+
+      # create output object
+      x <- NULL
+
+      # loop over bunches
+      for (b in seq_along(yearsets)) {
+        # read in a bunch of years
+        tmp <- lpjclass::read.LPJ_input(file_name = filename,
+                                        out_years = paste0("y", yearsets[[b]]),
+                                        namesum   = FALSE,
+                                        ncells    = 67420)
+        class(tmp) <- "array"
+        # KRISTINE: Please double-check whether following line makes sense
+        # (introduced because as.magpie doesn't work on 4-dimensional object)
+        tmp        <- abind::adrop(tmp, drop = 4)
+        tmp        <- as.magpie(tmp, spatial = 1)
+        getSets(tmp) <- c("fake", "year", "day")
+        # KRISTINE: Please double-check whether following line makes sense
+        # (introduced because toolAggregate doesn't work without dimension names)
+        getNames(tmp) <- as.character(seq(1, 365, 1))
+
+        # aggregate days to month
+        tmp <- toolAggregate(tmp,
+                             rel  = month2day,
+                             from = "day",
+                             to   = "month",
+                             dim  = 3)
+
+        if (subset == "monthlyMean") {
+          tmp <- tmp / monthLength
+        }
+
+        x <- mbind(x, tmp)
+        getSets(x) <- c("fake", "year", "month")
+      }
+
+    } else if (grepl("\\d{4}:\\d{4}", subset)) {
+
+      subYears <- eval(parse(text = subset))
+      years    <- intersect(years, subYears)
+      if (any(!(subYears %in% years))) {
+        warning(paste0("Some subsetted years (subset = ", subset,
+                       ") are not availabl\n in the original data.
+                       Years set to:", years))
```

Comment:
Mixing `\n` and actual newlines is a bit hard to read, you want the following, right?
```suggestion
        warning("Some subsetted years (subset = ", subset,
                ") are not availabl\n in the original data.\n",
                "Years set to:", years))
```